### PR TITLE
[serve] Stream Serve logs across different drivers

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Add back `_filter_logs_by_job` to worker.py, and use it to disable filtering of streamed logs in `print_logs`.
This existed in worker.py before, but was removed at some point.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -461,6 +461,8 @@ def __init__(self):
         # Create the lock here because the serializer will use it before
         # initializing Ray.
         self.lock = threading.RLock()
+        # By default, don't show logs from other drivers.
+        self._filter_logs_by_job = True
```
Comment (edoakes) [inline python/ray/_private/worker.py:465]: this is super hacky, so please update the comment to indicate why it's here and point at this PR :)

**Review by edoakes — COMMENTED** [review]
Unfortunate that we are resorting to the hack again but excellent that it'll be fixed :)

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/tests/test_standalone2.py [hunk 2]
```diff
@@ -419,6 +419,55 @@ def f():
...
+
+        f2.write(file2.encode("utf-8"))
+        f2.seek(0)
+        # Driver 2 (reconnects to the same Serve controller)
+        output = subprocess.check_output(["python", f2.name], stderr=subprocess.STDOUT)
+        assert "Adding 1 replica to deployment default_B" in output.decode("utf-8")
+
+    serve.shutdown()
+    ray.shutdown()
+    subprocess.check_output(["ray", "stop", "--force"])
```
Comment (edoakes) [inline python/ray/serve/tests/test_standalone2.py:468]: why are these necessary? shouldn't they be handled by the fixture?

Comment (zcin) [reply python/ray/serve/tests/test_standalone2.py:468]: Ray needs to be started through the cli, otherwise the serve scripts that are run in the test will start their own ray clusters and the test isn't meaningful. Let me pull out this code into a fixture!

File: python/ray/_private/worker.py [hunk 3]
```diff
@@ -462,6 +462,11 @@ def __init__(self):
         # Create the lock here because the serializer will use it before
         # initializing Ray.
         self.lock = threading.RLock()
+        # By default, don't show logs from other drivers. This is set to true by Serve
+        # in order to stream logs from the controller and replica actors across
+        # different drivers that connect to the same Serve instance.
+        # See https://github.com/ray-project/ray/pull/35070.
+        self._filter_logs_by_job = True
```
Comment (rkooo567) [inline python/ray/_private/worker.py:469]: Hm we may want to have a lock when you change this value.

**Review by rkooo567 — APPROVED** [review]
Looks like accessing boolean is already atomic, so it is protected by python thanks to GIL. LGTM

**Comment by rkooo567** [comment]
Lmk when it is ready to be merged

**Comment by edoakes** [comment]
maybe relevant test failure
